python_client/kubetorch/cli.py

```diff
@@ -1698,6 +1698,243 @@ def kt_volumes(
             raise typer.Exit(1)
 
 
+@app.command("notebook")
+def kt_notebook(
+    cpus: str = typer.Option(None, "--cpus", help="CPU resources (e.g., '2', '500m')"),
+    memory: str = typer.Option(None, "--memory", "-m", help="Memory resources (e.g., '4Gi', '512Mi')"),
+    gpus: str = typer.Option(None, "--gpus", help="Number of GPUs"),
+    image: str = typer.Option(None, "--image", "-i", help="Container image to use"),
+    name: str = typer.Option(None, "--name", help="Service name for the notebook"),
+    namespace: str = typer.Option(
+        globals.config.namespace,
+        "-n",
+        "--namespace",
+    ),
+    local_port: int = typer.Option(8888, "--port", "-p", help="Local port for notebook access"),
+    inactivity_ttl: str = typer.Option(None, "--ttl", help="Inactivity TTL (e.g., '1h', '30m')"),
+    restart_kernels: bool = typer.Option(
+        True,
+        "--restart/--no-restart",
+        help="Restart notebook kernel sessions upon reconnect",
+    ),
+):
+    """
+    Launch a JupyterLab notebook server on Kubernetes. The notebook service will continue running after you exit,
+    and you can reconnect to it until the service is torn down.
+
+    Examples:
+
+    .. code-block:: bash
+
+        $ kt notebook --cpus 4 --memory 8Gi # Launch with specific resources
+
+        $ kt notebook --gpus 1 --cpus 8 --memory 16Gi --image nvcr.io/nvidia/pytorch:23.10-py3  # Launch with GPU and custom image
+
+        $ kt notebook --gpus 1 --cpus 8 --memory 16Gi --no-restart # Don't restart kernels on reconnect
+    """
+    import webbrowser
+
+    import kubetorch as kt
+
+    if is_running_in_kubernetes():
+        console.print(
+            "[red]Notebook command is not supported when running inside Kubernetes. "
+            "Please run this command locally.[/red]"
+        )
+        raise typer.Exit(1)
+
+    # Build compute configuration
+    compute_kwargs = {"namespace": namespace}
+
+    if cpus:
+        compute_kwargs["cpus"] = cpus
+    if memory:
+        compute_kwargs["memory"] = memory
+    if gpus:
+        compute_kwargs["gpus"] = gpus
+    if inactivity_ttl:
+        compute_kwargs["inactivity_ttl"] = inactivity_ttl
```

Comment:
nit: the default `None` for these `if args:` is the same as the compute default so this could just be 

```
    compute_kwargs = {
        "namespace": namespace,
        "cpus": cpus,
        "memory": memory,
        "gpus": gpus,
        "inactivity_ttl": inactivity_ttl,
    }
```